engine/class_modules/paladin/sc_paladin_holy.cpp

```diff
@@ -450,16 +455,90 @@ struct lights_hammer_t : public paladin_spell_t
 
 // Light of Dawn (Holy) =======================================================
 
-struct light_of_dawn_t : public paladin_heal_t
+struct light_of_dawn_t : public holy_power_consumer_t<paladin_heal_t>
 {
   light_of_dawn_t( paladin_t* p, const std::string& options_str )
-    : paladin_heal_t( "light_of_dawn", p, p->find_class_spell( "Light of Dawn" ) )
+    : holy_power_consumer_t( "light_of_dawn", p, p->find_class_spell( "Light of Dawn" ) )
   {
     parse_options( options_str );
 
     aoe = 6;
+  }
+
+  void execute() override
+  {
+    holy_power_consumer_t::execute();
+
+    // deal with awakening
+    if ( p()->talents.awakening->ok() )
+    {
+      if ( rng().roll( p()->talents.awakening->effectN( 1 ).percent() ) )
+      {
+        buff_t* main_buff           = p()->buffs.avenging_wrath;
+        timespan_t trigger_duration = timespan_t::from_seconds( p()->talents.awakening->effectN( 2 ).base_value() );
+        if ( main_buff->check() )
+        {
+          p()->buffs.avengers_might->extend_duration( p(), trigger_duration );
+        }
+        else
+        {
+          main_buff->trigger( 1, buff_t::DEFAULT_VALUE(), -1.0, trigger_duration );
+        }
+      }
+    }
+  }
+};
+
+// Word Of Glory (Holy) =======================================================
+
+struct word_of_glory_t : public holy_power_consumer_t<paladin_heal_t>
+{
+  word_of_glory_t( paladin_t* p, const std::string& options_str )
+    : holy_power_consumer_t( "word_of_glory", p, p->find_class_spell( "Word Of Glory" ) )
+  {
+    parse_options( options_str );
+  }
+
+  void execute() override
+  {
+    holy_power_consumer_t::execute();
 
-    cooldown = p->cooldowns.light_of_dawn;
+    // deal with awakening
+    if ( p()->talents.awakening->ok() )
```

Comment:
It feels weird to me to do that. Since it only is a thing for holy paladin so if it was in holy_power_consumer_t or paladin_action_t it would just needlessly run for every ret and prot sim (the check for awakening even if you pre assign a struct wide bool for it.)

I feel like it being in heal could have the same result but lead the possibility to having a singular healing cast proc awakening multiple times

Reply:
The Idea is to add a function
```
void trigger_awakening();
```
to holy_power_consumer_t, paladin_heal_t or paladin_action_t

Only in word_of_glory_t::execute and light_of_dawn_t::execute are you then actually calling the function.

So the function is not triggered everywhere, the code is just shared in some appropriate base class.

Reply:
That makes a lot more sense. I thought you were just saying to generically put it in the parent execute.

Reply:
I kind of want to save this for stage 2 as right now its not messing anything up

Reply:
Sure no problem. The danger in copy-pasting code is that it quickly gets out of sync and later you might not always remember that the 2 places are actually meant to do the same thing ;)